### DISCUSSION TITLE
Just how many coffees does the StopForumSpam Dev drink?

### DISCUSSION THREAD

**Discussion by 737simpilot** [CIDRAM/CIDRAM#300] [General]
Well, I just did a rough estate based on four months worth of Wayback Machine data and it appears to be around ~ 3 cups per day. LOL!  His coffee drinking stat counter appears to have been added February, 2017. I plan on doing the math of that time till now. but I have a felling I'll arrive at ~3 cups per day like what I see already. I think I'll chart it as well.

![img](https://i.postimg.cc/zvTzyQfh/ezgif-com-gif-maker-16.gif)

On another subject. Can anyone teach this dunce what the exact differences in definition are for the words implicit and explicit? 

It appears if I'm being implicit I'm just implying, but it also means unconditional which is like explicit in my view because explicit means fully and clearly expressed. 

https://www.dictionary.com/browse/implicit

https://www.dictionary.com/browse/explicit

Now I have to see how this relates to the FTP protocol. I've read about it years ago and just don't remember.

**Comment by Maikuolan** [comment]
> On another subject. Can anyone teach this dunce what the exact differences in definition are for the words implicit and explicit?

Per the way I think of it, something "explicit" would be directly stated, clearly unveiled to the eyes, etc, whereas something "implicit" would be implied, not directly stated itself but nonetheless inferable from whatever else is otherwise directly stated, perhaps not clearly unveiled but nonetheless still discernible.

As an example in the context of CIDRAM: Let's say you discover some newly occurring malicious activity at your website, and after investigating it, determine that the cause is two distinct bots that'd recently started hammering various websites, including your own. Further investigating these two distinct bots, you discover that one of them *always* cites a specific, clear, unambiguous user agent, which you decide to use as the sole condition in a new auxiliary rule, in order to block that specific bot. You could say that you're *explicitly* blocking that particular bot, because your rule works on a singular, specific identifier which specifically targets that bot only and always.

You also discover that the *other* bot *never* cites a specific user agent, making it a little trickier to block, but also discover that its behaviour is somewhat predictable, rotating through a series of specific URLs or resources that it tries to request from your server, at specific time intervals, or perhaps making a series of odd requests from a range of cloud-based addresses which your standard human-based traffic should never originate from normally. It so happens that several dozen *other* bots *also* behave in the exact same way, but you're still reasonably sure that this is a new bot, because the timing of the newly discovered malicious activity and the timing of this new bot being released into the wild coincide, and there may be several very small, minor differences between the way this new bot behaves and the several dozen other bots behave, but these differences are small enough that the activities of these bots can only be distinguished from each other upon careful analysis of access logs as a whole, and only after the event. You decide to create a new auxiliary rule, targeting these specific behaviours that you've discovered, blocking requests originating from specific, suspicious origins, while requesting specific resources from your server which humans wouldn't normally request, perhaps even employing some heuristic controls, like blocking after a particular number of infractions have already been incurred during prior requests, build up upon behaviours detected at prior requests. This newly created rule is capable of blocking *all* of those several dozen bots in question, as well as the newly discovered bot, and does so effectively and efficiently, but isn't capable of distinguishing *which* of those several dozen bots in question are being blocked at any given block event. Accordingly, you could say that you're *implicitly* blocking that particular bot.

**Comment by Maikuolan** [comment]
> Now I have to see how this relates to the FTP protocol. I've read about it years ago and just don't remember.

In the context of FTP, I wouldn't put too much weight on the *linguistic* differences between "implicit" and "explicit", because in the context of FTP, it's actually referring to a difference in protocols utilised (SFTP versus FTPS), one which simply adds a layer on top of standard FTP, the other building upon SSH (so, not actually part of the FTP protocol at all). To avoid too much confusion, I would just treat it as merely referencing one or the other of the two, and just focus more on the actual differences between the two protocols instead, rather than the differences in the words used to describe them.

**Comment by Maikuolan** [comment]
> Well, I just did a rough estate based on four months worth of Wayback Machine data and it appears to be around ~ 3 cups per day.

Sounds like me, lol. Probably not something to brag about too much though. '^^

**Comment by 737simpilot** [comment]
Understand now the differences between implicit and explicit. Thanks.

Just found the answer on implicit and explicit FTP. https://en.wikipedia.org/wiki/FTPS#Methods_of_invoking_security


>Sounds like me, lol. Probably not something to brag about too much though. '^^

I'm the same. But more like 5 cups a day. LOL! Although, I drink decaffeinated which is like .3% caffeine per the whole bag of coffee. So per volume in the K-cup I use in the brewer the caffeine content is probably in the .0xx percentage range per cup of coffee I guess.  This is what I'm going by on the bag. The bag says something like 99.7 % caffeine free per bag or unit. I thought that was really interesting because I thought a whole bag of decaffeinated would be a bit more. Like 2% caffeine content ( a whole number as opposed to a fraction). I drink something called Gevalia. House bled, mild roast to be specific. LOL I used to have a nice espresso machine and made lattes with a really great coffee here you can buy in the states called Cafe Bustelo. It comes in a can and is in powdered form perfect for espresso machines. I made so many lattes I should have bought a cow! LOL! That's a real problem with lattes. Half is milk and half is coffee. 

https://en.wikipedia.org/wiki/Caf%C3%A9_Bustelo

https://en.wikipedia.org/wiki/Gevalia


It's unreal how one company is a subsidy of another and what not. https://en.wikipedia.org/wiki/Kraft_Heinz


Heinz is owned by multi national conglomerate Berkshire Hathaway, and Berkshire Hathaway is run by multi billionaire Warren Buffet. LOL! 

Interesting reading on _Tenbucks_ https://en.wikipedia.org/wiki/Starbucks


PS:

I hear at least two cups of coffee is good for you. Not sure about it all and whether that has to be decaffeinated or not.
